test/integration/tlsroute_test.go

```diff
@@ -356,11 +463,293 @@ func TestTLSRouteEssentials(t *testing.T) {
 	t.Log("verifying that the data-plane configuration from the TLSRoute does not get orphaned with the GatewayClass and Gateway gone")
 	require.Eventually(t, func() bool {
 		responded, err := tlsEchoResponds(fmt.Sprintf("%s:%d", proxyURL.Hostname(), ktfkong.DefaultTLSServicePort),
-			testUUID1, tlsRouteHostname)
+			testUUID, tlsRouteHostname, tlsRouteHostname)
 		return responded == false && errors.Is(err, io.EOF)
 	}, ingressWait, waitTick)
```

Comment:
> Verify in the TLS chain valid sense? We don't care if a cert is trusted given a set of CAs

No sorry, not this.

> just that Kong serves the certificate that we want it to

This. Did I miss it are we already doing this somewhere in here?

Reply:
If Kong did not load some certificate for that hostname, it will serve its default `*  subject: C=US; ST=California; L=San Francisco; O=Kong; OU=IT Department; CN=localhost` certificate. The CN check here confirms we're instead getting a certificate with the hostname we expect. This PR includes changes to `tlsEchoResponds()` to check the second hostname argument against the CN.

This isn't 100% foolproof because you could technically get some other certificate with `CN = tlsRouteHostname`, we just currently rely on the test environment conditions to not create that scenario

Reply:
:+1: